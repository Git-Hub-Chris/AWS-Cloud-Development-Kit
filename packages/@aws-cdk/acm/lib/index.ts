export * from './certificate';
export * from './certificate-ref';

// The L1 library for AWS::CertificateManager:
export * from '../cfn/certificatemanager';
