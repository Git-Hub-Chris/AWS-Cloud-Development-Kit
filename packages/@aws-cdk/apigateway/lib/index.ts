// The L1 Library for AWS::ApiGateway:
export * from '../cfn/apigateway';
