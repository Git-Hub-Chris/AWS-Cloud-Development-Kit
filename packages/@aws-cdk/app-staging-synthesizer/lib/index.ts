export * from './default-staging-stack';
export * from './synthesizer';
