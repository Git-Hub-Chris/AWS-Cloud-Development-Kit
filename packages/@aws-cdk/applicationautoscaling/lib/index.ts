// The L1 Library for AWS::ApplicationAutoScaling:
export * from '../cfn/applicationautoscaling';
