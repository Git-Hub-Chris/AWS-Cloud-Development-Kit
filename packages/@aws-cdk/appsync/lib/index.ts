// The L1 Library for AWS::AppSync:
export * from '../cfn/appsync';
