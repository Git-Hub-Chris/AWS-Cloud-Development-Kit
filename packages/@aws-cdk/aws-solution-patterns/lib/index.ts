export * from './lambda-dynamodb/index';