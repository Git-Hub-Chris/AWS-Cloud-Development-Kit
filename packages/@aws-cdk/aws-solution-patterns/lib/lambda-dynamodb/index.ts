import dynamodb = require('@aws-cdk/aws-dynamodb');
import lambda = require('@aws-cdk/aws-lambda');
import { Construct } from '@aws-cdk/core';
import { DefaultTableProps } from '../core/dynamodb-table-defaults';
import { DefaultFunctionProps } from '../core/lambda-defaults';
import { overrideProps } from '../core/utils';

export interface LambdaToDynamoDBProps {
  /**
   * Whether to create a new lambda function or use an existing lambda function
   *
   * If set to false, you must provide a lambda function object as `existingObj`
   *
   * @default true
   */
  readonly deployLambda: boolean,
  /**
   * [disable-awslint:ref-via-interface]
   * Existing instance of Lambda Function object
   *
   * If `deploy` is set to false only then this property is required
   *
   * @default - None
   */
  readonly existingLambdaObj?: lambda.Function,
  /**
   * Optional user provided props to override the default props
   *
   * If `deploy` is set to true only then this property is required
   *
   * @default - Default props are used
   */
  readonly lambdaFunctionProps?: lambda.FunctionProps,
  /**
   * Optional user provided props to override the default props
   *
   * @default - Default props are used
   */
  readonly dynamoTableProps?: dynamodb.TableProps
}

export class LambdaToDynamoDB extends Construct {
  private fn: lambda.Function;
  private table: dynamodb.Table;

  /**
   * @param {cdk.App} scope
   * @param {string} id
   * @param {cdk.StackProps=} props
   */
  constructor(scope: Construct, id: string, props: LambdaToDynamoDBProps) {
    super(scope, id);

    // Set the default props for DynamoDB table
    if (props.dynamoTableProps) {
      const dynamoTableProps = overrideProps(DefaultTableProps, props.dynamoTableProps);
      this.table = new dynamodb.Table(this, 'DynamoTable', dynamoTableProps);
    } else {
      this.table = new dynamodb.Table(this, 'DynamoTable', DefaultTableProps);
    }

    // Conditional lambda function creation
    // If deployLambda == false
    if (props.hasOwnProperty('deployLambda') && props.deployLambda === false) {
      if (props.existingLambdaObj) {
        this.fn = props.existingLambdaObj;
      } else {
        throw Error('Missing existingLambdaObj from props for deploy = false');
      }
    // If deployLambda == true
    } else {
      if (props.lambdaFunctionProps) {
        const lambdaFunctionProps = overrideProps(DefaultFunctionProps, props.lambdaFunctionProps);
        this.fn = new lambda.Function(this, 'LambdaFunction', lambdaFunctionProps);
      } else {
        throw Error('Missing lambdaFunctionProps from props for deploy = true');
      }
    }

    this.fn.addEnvironment('DDB_TABLE_NAME', this.table.tableName);

    this.table.grantReadWriteData(this.fn.grantPrincipal);
  }

  public dynamoTable(): dynamodb.Table {
    return this.table;
  }

  public lambdaFunction(): lambda.Function {
    return this.fn;
  }
}