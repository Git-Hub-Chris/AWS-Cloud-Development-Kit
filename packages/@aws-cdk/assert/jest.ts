import { Stack } from "@aws-cdk/cdk";
import { SynthesizedStack } from "@aws-cdk/cx-api";
import { haveResource, ResourcePart } from "./lib/assertions/have-resource";
import { expect as ourExpect } from './lib/expect';

declare global {
  namespace jest {
    interface Matchers<R> {
      toHaveResource(resourceType: string,
                     properties?: any,
                     comparison?: ResourcePart,
                     allowValueExtension?: boolean): R;
    }
  }
}

expect.extend({
  toHaveResource(actual: SynthesizedStack | Stack,
                 resourceType: string,
                 properties?: any,
                 comparison?: ResourcePart,
                 allowValueExtension: boolean = false) {

    const assertion = haveResource(resourceType, properties, comparison, allowValueExtension);
    const inspector = ourExpect(actual);
    const pass = assertion.assertUsing(inspector);
    if (pass) {
      return {
        pass,
        message: `Expected ${JSON.stringify(inspector.value, null, 2)} not to match ${assertion.description}`,
      };
    } else {
      return {
        pass,
        message: `Expected ${JSON.stringify(inspector.value, null, 2)} to match ${assertion.description}`,
      };
    }
  }
});