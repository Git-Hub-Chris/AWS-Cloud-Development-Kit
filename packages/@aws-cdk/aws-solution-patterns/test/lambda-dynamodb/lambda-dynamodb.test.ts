import { SynthUtils } from '@aws-cdk/assert';
import '@aws-cdk/assert/jest';
import dynamodb = require('@aws-cdk/aws-dynamodb');
import lambda = require('@aws-cdk/aws-lambda');
import cdk = require("@aws-cdk/core");
import path = require('path');
import { LambdaToDynamoDB, LambdaToDynamoDBProps } from "../../lib";

function deployNewFunc(stack: cdk.Stack) {
  const props: LambdaToDynamoDBProps = {
    deployLambda: true,
    lambdaFunctionProps: {
      code: lambda.Code.fromAsset(path.join(__dirname, 'lambda/handler.zip')),
      runtime: lambda.Runtime.NODEJS_10_X,
      handler: 'index.handler'
    },
  };
  return new LambdaToDynamoDB(stack, 'test-lambda-dynamodb-stack', props);
}

function useExistingFunc(stack: cdk.Stack) {
  const lambdaFunctionProps: lambda.FunctionProps = {
    code: lambda.Code.fromAsset(path.join(__dirname, 'lambda/handler.zip')),
    runtime: lambda.Runtime.NODEJS_10_X,
    handler: 'index.handler'
  };

  const props: LambdaToDynamoDBProps = {
    deployLambda: false,
    existingLambdaObj: new lambda.Function(stack, 'MyExistingFunction', lambdaFunctionProps),
    dynamoTableProps: {
      billingMode: dynamodb.BillingMode.PROVISIONED,
      readCapacity: 3,
      writeCapacity: 3,
      partitionKey: {
          name: 'id',
          type: dynamodb.AttributeType.STRING
      }
    },
  };

  return new LambdaToDynamoDB(stack, 'test-lambda-dynamodb-stack', props);
}

test('snapshot test LambdaToDynamoDB default params', () => {
  const stack = new cdk.Stack();
  deployNewFunc(stack);
  expect(SynthUtils.toCloudFormation(stack)).toMatchSnapshot();
});

test('check lambda function properties for deploy: true', () => {
  const stack = new cdk.Stack();

  deployNewFunc(stack);

  expect(stack).toHaveResource('AWS::Lambda::Function', {
    Handler: "index.handler",
    Role: {
      "Fn::GetAtt": [
        "testlambdadynamodbstackLambdaFunctionServiceRole226532E1",
        "Arn"
      ]
    },
    Runtime: "nodejs10.x",
    Environment: {
      Variables: {
        LOGLEVEL: "INFO",
        DDB_TABLE_NAME: {
        Ref: "testlambdadynamodbstackDynamoTable8138E93B"
        }
      }
    }
  });
});
test('check dynamo table properties for deploy: true', () => {
  const stack = new cdk.Stack();

  deployNewFunc(stack);

  expect(stack).toHaveResource('AWS::DynamoDB::Table', {
    KeySchema: [
      {
        AttributeName: "id",
        KeyType: "HASH"
      }
    ],
    AttributeDefinitions: [
      {
        AttributeName: "id",
        AttributeType: "S"
      }
    ],
    ProvisionedThroughput: {
      ReadCapacityUnits: 5,
      WriteCapacityUnits: 5
    }
  });
});
test('check iot lambda function role for deploy: true', () => {
  const stack = new cdk.Stack();

  deployNewFunc(stack);

  expect(stack).toHaveResource('AWS::IAM::Role', {
    AssumeRolePolicyDocument: {
      Statement: [
        {
          Action: "sts:AssumeRole",
          Effect: "Allow",
          Principal: {
            Service: "lambda.amazonaws.com"
          }
        }
      ],
      Version: "2012-10-17"
    },
    ManagedPolicyArns: [
      {
        "Fn::Join": [
          "",
          [
            "arn:",
            {
              Ref: "AWS::Partition"
            },
            ":iam::aws:policy/service-role/AWSLambdaBasicExecutionRole"
          ]
        ]
      }
    ]
  });
});
test('check lambda function policy for deploy: true', () => {
  const stack = new cdk.Stack();

  deployNewFunc(stack);

  expect(stack).toHaveResource('AWS::IAM::Policy', {
    PolicyDocument: {
      Statement: [
        {
          Action: [
            "dynamodb:BatchGetItem",
            "dynamodb:GetRecords",
            "dynamodb:GetShardIterator",
            "dynamodb:Query",
            "dynamodb:GetItem",
            "dynamodb:Scan",
            "dynamodb:BatchWriteItem",
            "dynamodb:PutItem",
            "dynamodb:UpdateItem",
            "dynamodb:DeleteItem"
          ],
          Effect: "Allow",
          Resource: [
            {
              "Fn::GetAtt": [
                "testlambdadynamodbstackDynamoTable8138E93B",
                "Arn"
              ]
            },
            {
              Ref: "AWS::NoValue"
            }
          ]
        }
      ],
      Version: "2012-10-17"
    }
  });
});
test('check lambda function properties for deploy: false', () => {
  const stack = new cdk.Stack();

  useExistingFunc(stack);

  expect(stack).toHaveResource('AWS::Lambda::Function', {
      Handler: "index.handler",
      Role: {
        "Fn::GetAtt": [
          "MyExistingFunctionServiceRoleF9E14BFD",
          "Arn"
        ]
      },
      Runtime: "nodejs10.x"
  });
});
test('check iot lambda function role for deploy: false', () => {
  const stack = new cdk.Stack();

  useExistingFunc(stack);

  expect(stack).toHaveResource('AWS::IAM::Role', {
    AssumeRolePolicyDocument: {
      Statement: [
        {
          Action: "sts:AssumeRole",
          Effect: "Allow",
          Principal: {
            Service: "lambda.amazonaws.com"
          }
        }
      ],
      Version: "2012-10-17"
    },
    ManagedPolicyArns: [
      {
        "Fn::Join": [
          "",
          [
            "arn:",
            {
              Ref: "AWS::Partition"
            },
            ":iam::aws:policy/service-role/AWSLambdaBasicExecutionRole"
          ]
        ]
      }
    ]
  });
});
test('check getter methods', () => {
  const stack = new cdk.Stack();

  const construct: LambdaToDynamoDB = deployNewFunc(stack);

  expect(construct.lambdaFunction()).toBeInstanceOf(lambda.Function);
  expect(construct.dynamoTable()).toBeInstanceOf(dynamodb.Table);
});
test('check exception for Missing existingObj from props for deploy = false', () => {
  const stack = new cdk.Stack();

  const props: LambdaToDynamoDBProps = {
    deployLambda: false
  };

  try {
    new LambdaToDynamoDB(stack, 'test-iot-lambda-integration', props);
  } catch (e) {
    expect(e).toBeInstanceOf(Error);
  }
});
test('check exception for Missing lambdaFunctionProps from props for deploy = true', () => {
  const stack = new cdk.Stack();

  const props: LambdaToDynamoDBProps = {
    deployLambda: true
  };

  try {
    new LambdaToDynamoDB(stack, 'test-iot-lambda-integration', props);
  } catch (e) {
    expect(e).toBeInstanceOf(Error);
  }
});
