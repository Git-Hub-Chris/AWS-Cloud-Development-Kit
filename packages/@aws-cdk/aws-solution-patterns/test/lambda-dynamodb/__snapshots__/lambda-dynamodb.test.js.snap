// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot test LambdaToDynamoDB default params 1`] = `
Object {
  "Parameters": Object {
    "AssetParameters78e4557b6af07a75c658312f6045a0bb11a8ed0ed528857b3c0687477658ec9bArtifactHashBE359D28": Object {
      "Description": "Artifact hash for asset \\"78e4557b6af07a75c658312f6045a0bb11a8ed0ed528857b3c0687477658ec9b\\"",
      "Type": "String",
    },
    "AssetParameters78e4557b6af07a75c658312f6045a0bb11a8ed0ed528857b3c0687477658ec9bS3Bucket8D2850DA": Object {
      "Description": "S3 bucket for asset \\"78e4557b6af07a75c658312f6045a0bb11a8ed0ed528857b3c0687477658ec9b\\"",
      "Type": "String",
    },
    "AssetParameters78e4557b6af07a75c658312f6045a0bb11a8ed0ed528857b3c0687477658ec9bS3VersionKeyB5B23FDF": Object {
      "Description": "S3 key for asset version \\"78e4557b6af07a75c658312f6045a0bb11a8ed0ed528857b3c0687477658ec9b\\"",
      "Type": "String",
    },
    "AssetParametersb23e818e172b6771c5f4fde08381b8d0904b5760fcfcceee68523a83fff99c42ArtifactHashD6C3B3DD": Object {
      "Description": "Artifact hash for asset \\"b23e818e172b6771c5f4fde08381b8d0904b5760fcfcceee68523a83fff99c42\\"",
      "Type": "String",
    },
    "AssetParametersb23e818e172b6771c5f4fde08381b8d0904b5760fcfcceee68523a83fff99c42S3Bucket773E39E4": Object {
      "Description": "S3 bucket for asset \\"b23e818e172b6771c5f4fde08381b8d0904b5760fcfcceee68523a83fff99c42\\"",
      "Type": "String",
    },
    "AssetParametersb23e818e172b6771c5f4fde08381b8d0904b5760fcfcceee68523a83fff99c42S3VersionKey96D611D4": Object {
      "Description": "S3 key for asset version \\"b23e818e172b6771c5f4fde08381b8d0904b5760fcfcceee68523a83fff99c42\\"",
      "Type": "String",
    },
  },
  "Resources": Object {
    "LogRetentionaae0aa3c5b4d4f87b02d85b201efdd8aFD4BFC8A": Object {
      "DependsOn": Array [
        "LogRetentionaae0aa3c5b4d4f87b02d85b201efdd8aServiceRoleDefaultPolicyADDA7DEB",
        "LogRetentionaae0aa3c5b4d4f87b02d85b201efdd8aServiceRole9741ECFB",
      ],
      "Properties": Object {
        "Code": Object {
          "S3Bucket": Object {
            "Ref": "AssetParametersb23e818e172b6771c5f4fde08381b8d0904b5760fcfcceee68523a83fff99c42S3Bucket773E39E4",
          },
          "S3Key": Object {
            "Fn::Join": Array [
              "",
              Array [
                Object {
                  "Fn::Select": Array [
                    0,
                    Object {
                      "Fn::Split": Array [
                        "||",
                        Object {
                          "Ref": "AssetParametersb23e818e172b6771c5f4fde08381b8d0904b5760fcfcceee68523a83fff99c42S3VersionKey96D611D4",
                        },
                      ],
                    },
                  ],
                },
                Object {
                  "Fn::Select": Array [
                    1,
                    Object {
                      "Fn::Split": Array [
                        "||",
                        Object {
                          "Ref": "AssetParametersb23e818e172b6771c5f4fde08381b8d0904b5760fcfcceee68523a83fff99c42S3VersionKey96D611D4",
                        },
                      ],
                    },
                  ],
                },
              ],
            ],
          },
        },
        "Handler": "index.handler",
        "Role": Object {
          "Fn::GetAtt": Array [
            "LogRetentionaae0aa3c5b4d4f87b02d85b201efdd8aServiceRole9741ECFB",
            "Arn",
          ],
        },
        "Runtime": "nodejs10.x",
      },
      "Type": "AWS::Lambda::Function",
    },
    "LogRetentionaae0aa3c5b4d4f87b02d85b201efdd8aServiceRole9741ECFB": Object {
      "Properties": Object {
        "AssumeRolePolicyDocument": Object {
          "Statement": Array [
            Object {
              "Action": "sts:AssumeRole",
              "Effect": "Allow",
              "Principal": Object {
                "Service": "lambda.amazonaws.com",
              },
            },
          ],
          "Version": "2012-10-17",
        },
        "ManagedPolicyArns": Array [
          Object {
            "Fn::Join": Array [
              "",
              Array [
                "arn:",
                Object {
                  "Ref": "AWS::Partition",
                },
                ":iam::aws:policy/service-role/AWSLambdaBasicExecutionRole",
              ],
            ],
          },
        ],
      },
      "Type": "AWS::IAM::Role",
    },
    "LogRetentionaae0aa3c5b4d4f87b02d85b201efdd8aServiceRoleDefaultPolicyADDA7DEB": Object {
      "Properties": Object {
        "PolicyDocument": Object {
          "Statement": Array [
            Object {
              "Action": Array [
                "logs:PutRetentionPolicy",
                "logs:DeleteRetentionPolicy",
              ],
              "Effect": "Allow",
              "Resource": "*",
            },
          ],
          "Version": "2012-10-17",
        },
        "PolicyName": "LogRetentionaae0aa3c5b4d4f87b02d85b201efdd8aServiceRoleDefaultPolicyADDA7DEB",
        "Roles": Array [
          Object {
            "Ref": "LogRetentionaae0aa3c5b4d4f87b02d85b201efdd8aServiceRole9741ECFB",
          },
        ],
      },
      "Type": "AWS::IAM::Policy",
    },
    "testlambdadynamodbstackDynamoTable8138E93B": Object {
      "DeletionPolicy": "Retain",
      "Properties": Object {
        "AttributeDefinitions": Array [
          Object {
            "AttributeName": "id",
            "AttributeType": "S",
          },
        ],
        "KeySchema": Array [
          Object {
            "AttributeName": "id",
            "KeyType": "HASH",
          },
        ],
        "ProvisionedThroughput": Object {
          "ReadCapacityUnits": 5,
          "WriteCapacityUnits": 5,
        },
      },
      "Type": "AWS::DynamoDB::Table",
      "UpdateReplacePolicy": "Retain",
    },
    "testlambdadynamodbstackLambdaFunction5DDB3E8D": Object {
      "DependsOn": Array [
        "testlambdadynamodbstackLambdaFunctionServiceRoleDefaultPolicyF537063F",
        "testlambdadynamodbstackLambdaFunctionServiceRole226532E1",
      ],
      "Properties": Object {
        "Code": Object {
          "S3Bucket": Object {
            "Ref": "AssetParameters78e4557b6af07a75c658312f6045a0bb11a8ed0ed528857b3c0687477658ec9bS3Bucket8D2850DA",
          },
          "S3Key": Object {
            "Fn::Join": Array [
              "",
              Array [
                Object {
                  "Fn::Select": Array [
                    0,
                    Object {
                      "Fn::Split": Array [
                        "||",
                        Object {
                          "Ref": "AssetParameters78e4557b6af07a75c658312f6045a0bb11a8ed0ed528857b3c0687477658ec9bS3VersionKeyB5B23FDF",
                        },
                      ],
                    },
                  ],
                },
                Object {
                  "Fn::Select": Array [
                    1,
                    Object {
                      "Fn::Split": Array [
                        "||",
                        Object {
                          "Ref": "AssetParameters78e4557b6af07a75c658312f6045a0bb11a8ed0ed528857b3c0687477658ec9bS3VersionKeyB5B23FDF",
                        },
                      ],
                    },
                  ],
                },
              ],
            ],
          },
        },
        "Environment": Object {
          "Variables": Object {
            "DDB_TABLE_NAME": Object {
              "Ref": "testlambdadynamodbstackDynamoTable8138E93B",
            },
            "LOGLEVEL": "INFO",
          },
        },
        "Handler": "index.handler",
        "Role": Object {
          "Fn::GetAtt": Array [
            "testlambdadynamodbstackLambdaFunctionServiceRole226532E1",
            "Arn",
          ],
        },
        "Runtime": "nodejs10.x",
      },
      "Type": "AWS::Lambda::Function",
    },
    "testlambdadynamodbstackLambdaFunctionLogRetention4724F084": Object {
      "Properties": Object {
        "LogGroupName": Object {
          "Fn::Join": Array [
            "",
            Array [
              "/aws/lambda/",
              Object {
                "Ref": "testlambdadynamodbstackLambdaFunction5DDB3E8D",
              },
            ],
          ],
        },
        "RetentionInDays": 90,
        "ServiceToken": Object {
          "Fn::GetAtt": Array [
            "LogRetentionaae0aa3c5b4d4f87b02d85b201efdd8aFD4BFC8A",
            "Arn",
          ],
        },
      },
      "Type": "Custom::LogRetention",
    },
    "testlambdadynamodbstackLambdaFunctionServiceRole226532E1": Object {
      "Properties": Object {
        "AssumeRolePolicyDocument": Object {
          "Statement": Array [
            Object {
              "Action": "sts:AssumeRole",
              "Effect": "Allow",
              "Principal": Object {
                "Service": "lambda.amazonaws.com",
              },
            },
          ],
          "Version": "2012-10-17",
        },
        "ManagedPolicyArns": Array [
          Object {
            "Fn::Join": Array [
              "",
              Array [
                "arn:",
                Object {
                  "Ref": "AWS::Partition",
                },
                ":iam::aws:policy/service-role/AWSLambdaBasicExecutionRole",
              ],
            ],
          },
        ],
      },
      "Type": "AWS::IAM::Role",
    },
    "testlambdadynamodbstackLambdaFunctionServiceRoleDefaultPolicyF537063F": Object {
      "Properties": Object {
        "PolicyDocument": Object {
          "Statement": Array [
            Object {
              "Action": Array [
                "dynamodb:BatchGetItem",
                "dynamodb:GetRecords",
                "dynamodb:GetShardIterator",
                "dynamodb:Query",
                "dynamodb:GetItem",
                "dynamodb:Scan",
                "dynamodb:BatchWriteItem",
                "dynamodb:PutItem",
                "dynamodb:UpdateItem",
                "dynamodb:DeleteItem",
              ],
              "Effect": "Allow",
              "Resource": Array [
                Object {
                  "Fn::GetAtt": Array [
                    "testlambdadynamodbstackDynamoTable8138E93B",
                    "Arn",
                  ],
                },
                Object {
                  "Ref": "AWS::NoValue",
                },
              ],
            },
          ],
          "Version": "2012-10-17",
        },
        "PolicyName": "testlambdadynamodbstackLambdaFunctionServiceRoleDefaultPolicyF537063F",
        "Roles": Array [
          Object {
            "Ref": "testlambdadynamodbstackLambdaFunctionServiceRole226532E1",
          },
        ],
      },
      "Type": "AWS::IAM::Policy",
    },
  },
}
`;
